onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -2566,6 +2594,32 @@ static bool CastNeedFallbackToCPU(const onnxruntime::Node& node) {
   return false;
 }
 
+static bool ArgMaxOrArgMinNeedFallbackToCPU(const onnxruntime::Node& node) {
+  // Opset 12 introduced the attribute "select_last_index"
+  if (node.SinceVersion() >= 12) {
+    const auto& node_attributes = node.GetAttributes();
+
+    for (auto& attr : node_attributes) {
+      auto& attr_name = attr.first;
+      auto& attr_value = attr.second;
+
+      // CuDNN doesn't support picking the last index in case of encountering
+      // duplicate max values.
+      // CuDNN's API doc doesn't mention what happens in case duplicates are encountered,
+      // but based on testing, the results seem to indicate a "stable" implementation
```

Comment:
Could you please double-check if the first index is always returned by the CuDNN implementation in case of duplicates (as required by the spec in case select_last_index == 0) ? I wasn't able to confirm this from the API docs at that time. Even though testing seems to show that it always picked the first, I wasn't sure if we were just getting lucky with the tests.

Reply:
It's true based on my tests. I've updated the test using random index for verification.